### ISSUE TITLE
h2o/quicly panics

### ISSUE THREAD

**Issue by agrover** [mozilla/neqo#315]
`RUST_BACKTRACE=1 RUST_LOG=neqo_transport::dump ./target/debug/neqo-client --omit-read-data https://h2o.examp1e.net:443/`

```
0s193ms DEBUG [Client 0x7ffcddce2888] pn=3 type=Handshake
  -> RX Crypto { offset: 2343, len: 625 }
thread 'main' panicked at 'assertion failed: `(left == right)`
  left: `Client`,
 right: `Server`', neqo-transport/src/connection.rs:869:13
stack backtrace:
   0: <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt
   1: core::fmt::write
   2: std::io::Write::write_fmt
   3: std::panicking::default_hook::{{closure}}
   4: std::panicking::default_hook
   5: std::panicking::rust_panic_with_hook
   6: std::panicking::continue_panic_fmt
   7: std::panicking::begin_panic_fmt
   8: neqo_transport::connection::Connection::decrypt_body
             at neqo-transport/src/connection.rs:869
   9: neqo_transport::connection::Connection::input
             at neqo-transport/src/connection.rs:841
  10: neqo_transport::connection::Connection::process_input
```

**Comment by ddragana** [comment]
I moved the check and the assertion does not hold any more. I will fix it.

**Comment by ddragana** [comment]
Fixed by #321
